### PR TITLE
Enhancing Model Accuracy through Advanced Transfer Learning Technique

### PR DESCRIPTION
**Pull request by DarshAgrawal14**
Issue : https://github.com/Akasxh/Terrain-Recognition/issues/8
- The current implementation utilizes EfficientNet for terrain classification.
- Data Agumentation includes resizing, horizontal flipping, color jitter, and random rotations.
- I have loaded a pre-trained EfficientNet B0 model and modified the classifier layer to match the number of output classes (5 for different terrain types).
- I noticed at high learning rates the accuracy was highly volatile so it was lowered
- I defined the loss function (CrossEntropyLoss) and optimizer (Adam) with a lower learning rate for fine-tuning.
- Learning rate scheduling : ReduceLROnPlateau
- Included code to load model and test it using a example image
- The hyper - parameter tuning was performed using gpu , but if not available code will automatically use cpu
- Included code to load model and test it using a example image
- The hyper - parameter tuning was performed using gpu , but if not available code will automatically use cpu
- added tranfer learning notebooks , model , Readme.md ,improvement.md in transfer_learning directory
-  Updated main README.md

Darsh Agrawal , GSSoC 2024 extd contributor

### CODE REVIEW

**Comment by DarshAgrawal14** [comment]
@Akasxh it seems the corrects labels are hactoberfest , gssoc-ext , level 3 , can you please update the labels so that the associated points are awarded.

**Comment by AKSHITHA-CHILUKA** [comment]
@Akasxh despite of many times telling you have not changed the labels.
The crt labels are level3 with no spaces , gssoc-ext is the label not gssoc24-extd , hactoberfest-accepted is the label not hactoberfest .
I hope you will change all the label's in all the issues and pr's as soon as possible .

**Comment by Akasxh** [comment]
@AKSHITHA-CHILUKA I am not available 24/7, I have updated the labels as soon I got back to my machine.

**Comment by AKSHITHA-CHILUKA** [comment]
so now you are available so I will see now whether you have added all the required labels or not

**Comment by Akasxh** [comment]
I have updated all the labels please check if there's any issue.

**Comment by AKSHITHA-CHILUKA** [comment]
image
All the issues and pr should be labeled

**Comment by Akasxh** [comment]
Should I be assigning once it's completed or just as a issue is raised? 

The points are taken from prs right?

**Comment by AKSHITHA-CHILUKA** [comment]
both issue and pr if and only if both are labeled , then only the points will be reflected ,once you assign them you have to add all the labels including level labels .

**Comment by Akasxh** [comment]
Okay done
image

**Comment by AKSHITHA-CHILUKA** [comment]
ok for further doubts please connect with @Meetjain1
